File: str_helper.cpp
// system includes
#include <map>

// includes
#include "str_helper.h"
#include "shopndrop_protocol/str_helper.h"
#include "generic_protocol/str_helper.h"
#include "basic_objects/str_helper.h"
#include "basic_parser/str_helper.h"

namespace shopndrop_web_protocol
{

namespace str_helper
{

using ::basic_parser::str_helper::write;
using ::basic_parser::str_helper::write_t;

// enums

#define TUPLE_VAL_STR(_x_)  _x_,#_x_

// objects

std::ostream & write( std::ostream & os, const ProductItemWithId & r )
{
    os << "(";

    os << " product_item_id="; write( os, r.product_item_id );
    os << " product_item="; ::shopndrop_protocol::str_helper::write( os, r.product_item );

    os << ")";

    return os;
}

std::ostream & write( std::ostream & os, const ShoppingItemWithProduct & r )
{
    os << "(";

    os << " shopping_item="; ::shopndrop_protocol::str_helper::write( os, r.shopping_item );
    os << " product_item="; ::shopndrop_protocol::str_helper::write( os, r.product_item );

    os << ")";

    return os;
}

std::ostream & write( std::ostream & os, const ShoppingListWithProduct & r )
{
    os << "(";

    os << " items="; write_t( os, r.items, static_cast<std::ostream & (*)( std::ostream &, const ShoppingItemWithProduct &  )>( &write ) ); // Array

    os << ")";

    return os;
}

std::ostream & write( std::ostream & os, const ShoppingListWithTotals & r )
{
    os << "(";

    os << " shopping_list="; write( os, r.shopping_list );
    os << " price="; write( os, r.price );
    os << " weight="; write( os, r.weight );

    os << ")";

    return os;
}

std::ostream & write( std::ostream & os, const RideSummaryWithShopper & r )
{
    os << "(";

    os << " ride_id="; write( os, r.ride_id );
    os << " ride="; ::shopndrop_protocol::str_helper::write( os, r.ride );
    os << " shopper_name="; write( os, r.shopper_name );

    os << ")";

    return os;
}

std::ostream & write( std::ostream & os, const RideWithId & r )
{
    os << "(";

    os << " ride_id="; write( os, r.ride_id );
    os << " ride="; ::shopndrop_protocol::str_helper::write( os, r.ride );

    os << ")";

    return os;
}

std::ostream & write( std::ostream & os, const ShoppingRequestInfo & r )
{
    os << "(";

    os << " order_id="; write( os, r.order_id );
    os << " sum="; write( os, r.sum );
    os << " earning="; write( os, r.earning );
    os << " weight="; write( os, r.weight );
    os << " address="; ::shopndrop_protocol::str_helper::write( os, r.address );

    os << ")";

    return os;
}

std::ostream & write( std::ostream & os, const AcceptedOrderUser & r )
{
    os << "(";

    os << " order_id="; write( os, r.order_id );
    os << " delivery_time="; ::basic_objects::str_helper::write( os, r.delivery_time );
    os << " order="; ::shopndrop_protocol::str_helper::write( os, r.order );
    os << " sum="; write( os, r.sum );
    os << " shopper_name="; write( os, r.shopper_name );

    os << ")";

    return os;
}

std::ostream & write( std::ostream & os, const AcceptedOrderShopper & r )
{
    os << "(";

    os << " order_id="; write( os, r.order_id );
    os << " delivery_time="; ::basic_objects::str_helper::write( os, r.delivery_time );
    os << " order="; ::shopndrop_protocol::str_helper::write( os, r.order );
    os << " sum="; write( os, r.sum );
    os << " earning="; write( os, r.earning );
    os << " weight="; write( os, r.weight );

    os << ")";

    return os;
}

std::ostream & write( std::ostream & os, const DashScreenUser & r )
{
    os << "(";

    os << " current_time="; ::basic_objects::str_helper::write( os, r.current_time );
    os << " rides="; write_t( os, r.rides, static_cast<std::ostream & (*)( std::ostream &, const RideSummaryWithShopper &  )>( &write ) ); // Array
    os << " orders="; write_t( os, r.orders, static_cast<std::ostream & (*)( std::ostream &, const AcceptedOrderUser &  )>( &write ) ); // Array

    os << ")";

    return os;
}

std::ostream & write( std::ostream & os, const DashScreenShopper & r )
{
    os << "(";

    os << " current_time="; ::basic_objects::str_helper::write( os, r.current_time );
    os << " rides="; write_t( os, r.rides, static_cast<std::ostream & (*)( std::ostream &, const RideWithId &  )>( &write ) ); // Array
    os << " orders="; write_t( os, r.orders, static_cast<std::ostream & (*)( std::ostream &, const AcceptedOrderShopper &  )>( &write ) ); // Array

    os << ")";

    return os;
}

// base messages

// messages

std::ostream & write( std::ostream & os, const GetProductItemListRequest & r )
{
    // base class
    ::shopndrop_protocol::str_helper::write( os, static_cast<const shopndrop_protocol::Request&>( r ) );


    return os;
}

std::ostream & write( std::ostream & os, const GetProductItemListResponse & r )
{
    // base class
    ::generic_protocol::str_helper::write( os, static_cast<const generic_protocol::BackwardMessage&>( r ) );

    os << " product_items="; write_t( os, r.product_items, static_cast<std::ostream & (*)( std::ostream &, const ProductItemWithId &  )>( &write ) ); // Array

    return os;
}

std::ostream & write( std::ostream & os, const GetShoppingRequestInfoRequest & r )
{
    // base class
    ::shopndrop_protocol::str_helper::write( os, static_cast<const shopndrop_protocol::Request&>( r ) );

    os << " ride_id="; write( os, r.ride_id );

    return os;
}

std::ostream & write( std::ostream & os, const GetShoppingRequestInfoResponse & r )
{
    // base class
    ::generic_protocol::str_helper::write( os, static_cast<const generic_protocol::BackwardMessage&>( r ) );

    os << " requests="; write_t( os, r.requests, static_cast<std::ostream & (*)( std::ostream &, const ShoppingRequestInfo &  )>( &write ) ); // Array

    return os;
}

std::ostream & write( std::ostream & os, const GetShoppingListWithTotalsRequest & r )
{
    // base class
    ::shopndrop_protocol::str_helper::write( os, static_cast<const shopndrop_protocol::Request&>( r ) );

    os << " shopping_list_id="; write( os, r.shopping_list_id );

    return os;
}

std::ostream & write( std::ostream & os, const GetShoppingListWithTotalsResponse & r )
{
    // base class
    ::generic_protocol::str_helper::write( os, static_cast<const generic_protocol::BackwardMessage&>( r ) );

    os << " shopping_list="; write( os, r.shopping_list );

    return os;
}

std::ostream & write( std::ostream & os, const GetDashScreenUserRequest & r )
{
    // base class
    ::shopndrop_protocol::str_helper::write( os, static_cast<const shopndrop_protocol::Request&>( r ) );

    os << " user_id="; write( os, r.user_id );
    os << " position="; ::shopndrop_protocol::str_helper::write( os, r.position );

    return os;
}

std::ostream & write( std::ostream & os, const GetDashScreenUserResponse & r )
{
    // base class
    ::generic_protocol::str_helper::write( os, static_cast<const generic_protocol::BackwardMessage&>( r ) );

    os << " dash_screen="; write( os, r.dash_screen );

    return os;
}

std::ostream & write( std::ostream & os, const GetDashScreenShopperRequest & r )
{
    // base class
    ::shopndrop_protocol::str_helper::write( os, static_cast<const shopndrop_protocol::Request&>( r ) );

    os << " user_id="; write( os, r.user_id );
    os << " position="; ::shopndrop_protocol::str_helper::write( os, r.position );

    return os;
}

std::ostream & write( std::ostream & os, const GetDashScreenShopperResponse & r )
{
    // base class
    ::generic_protocol::str_helper::write( os, static_cast<const generic_protocol::BackwardMessage&>( r ) );

    os << " dash_screen="; write( os, r.dash_screen );

    return os;
}

} // namespace str_helper

} // namespace shopndrop_web_protocol

File: str_helper.php
<?php

namespace shopndrop_web_protocol;


// includes
require_once __DIR__.'/../shopndrop_protocol/str_helper.php';
require_once __DIR__.'/../generic_protocol/str_helper.php';
require_once __DIR__.'/../basic_objects/str_helper.php';
require_once __DIR__.'/../basic_parser/str_helper.php';

// enums

// objects

function to_string__ProductItemWithId( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " product_item_id=" . \basic_parser\to_string__int( $r->product_item_id );
    $res .= " product_item=" . \shopndrop_protocol\to_string__ProductItem( $r->product_item );

    $res .= ")";

    return $res;
}

function to_string__ShoppingItemWithProduct( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " shopping_item=" . \shopndrop_protocol\to_string__ShoppingItem( $r->shopping_item );
    $res .= " product_item=" . \shopndrop_protocol\to_string__ProductItem( $r->product_item );

    $res .= ")";

    return $res;
}

function to_string__ShoppingListWithProduct( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " items=" . \basic_parser\to_string__vector( $r->items, '\shopndrop_web_protocol\to_string__ShoppingItemWithProduct' ); // Array

    $res .= ")";

    return $res;
}

function to_string__ShoppingListWithTotals( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " shopping_list=" . to_string__ShoppingListWithProduct( $r->shopping_list );
    $res .= " price=" . \basic_parser\to_string__float( $r->price );
    $res .= " weight=" . \basic_parser\to_string__float( $r->weight );

    $res .= ")";

    return $res;
}

function to_string__RideSummaryWithShopper( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " ride_id=" . \basic_parser\to_string__int( $r->ride_id );
    $res .= " ride=" . \shopndrop_protocol\to_string__RideSummary( $r->ride );
    $res .= " shopper_name=" . \basic_parser\to_string__string( $r->shopper_name );

    $res .= ")";

    return $res;
}

function to_string__RideWithId( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " ride_id=" . \basic_parser\to_string__int( $r->ride_id );
    $res .= " ride=" . \shopndrop_protocol\to_string__Ride( $r->ride );

    $res .= ")";

    return $res;
}

function to_string__ShoppingRequestInfo( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " order_id=" . \basic_parser\to_string__int( $r->order_id );
    $res .= " sum=" . \basic_parser\to_string__float( $r->sum );
    $res .= " earning=" . \basic_parser\to_string__float( $r->earning );
    $res .= " weight=" . \basic_parser\to_string__float( $r->weight );
    $res .= " address=" . \shopndrop_protocol\to_string__Address( $r->address );

    $res .= ")";

    return $res;
}

function to_string__AcceptedOrderUser( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " order_id=" . \basic_parser\to_string__int( $r->order_id );
    $res .= " delivery_time=" . \basic_objects\to_string__LocalTime( $r->delivery_time );
    $res .= " order=" . \shopndrop_protocol\to_string__Order( $r->order );
    $res .= " sum=" . \basic_parser\to_string__float( $r->sum );
    $res .= " shopper_name=" . \basic_parser\to_string__string( $r->shopper_name );

    $res .= ")";

    return $res;
}

function to_string__AcceptedOrderShopper( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " order_id=" . \basic_parser\to_string__int( $r->order_id );
    $res .= " delivery_time=" . \basic_objects\to_string__LocalTime( $r->delivery_time );
    $res .= " order=" . \shopndrop_protocol\to_string__Order( $r->order );
    $res .= " sum=" . \basic_parser\to_string__float( $r->sum );
    $res .= " earning=" . \basic_parser\to_string__float( $r->earning );
    $res .= " weight=" . \basic_parser\to_string__float( $r->weight );

    $res .= ")";

    return $res;
}

function to_string__DashScreenUser( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " current_time=" . \basic_objects\to_string__LocalTime( $r->current_time );
    $res .= " rides=" . \basic_parser\to_string__vector( $r->rides, '\shopndrop_web_protocol\to_string__RideSummaryWithShopper' ); // Array
    $res .= " orders=" . \basic_parser\to_string__vector( $r->orders, '\shopndrop_web_protocol\to_string__AcceptedOrderUser' ); // Array

    $res .= ")";

    return $res;
}

function to_string__DashScreenShopper( & $r )
{
    $res = "";

    $res .= "(";

    $res .= " current_time=" . \basic_objects\to_string__LocalTime( $r->current_time );
    $res .= " rides=" . \basic_parser\to_string__vector( $r->rides, '\shopndrop_web_protocol\to_string__RideWithId' ); // Array
    $res .= " orders=" . \basic_parser\to_string__vector( $r->orders, '\shopndrop_web_protocol\to_string__AcceptedOrderShopper' ); // Array

    $res .= ")";

    return $res;
}

// base messages

// messages

function to_string__GetProductItemListRequest( & $r )
{
    $res = "";

    // base class
    $res .= \shopndrop_protocol\to_string__Request( $r );


    return $res;
}

function to_string__GetProductItemListResponse( & $r )
{
    $res = "";

    // base class
    $res .= \generic_protocol\to_string__BackwardMessage( $r );

    $res .= " product_items=" . \basic_parser\to_string__vector( $r->product_items, '\shopndrop_web_protocol\to_string__ProductItemWithId' ); // Array

    return $res;
}

function to_string__GetShoppingRequestInfoRequest( & $r )
{
    $res = "";

    // base class
    $res .= \shopndrop_protocol\to_string__Request( $r );

    $res .= " ride_id=" . \basic_parser\to_string__int( $r->ride_id );

    return $res;
}

function to_string__GetShoppingRequestInfoResponse( & $r )
{
    $res = "";

    // base class
    $res .= \generic_protocol\to_string__BackwardMessage( $r );

    $res .= " requests=" . \basic_parser\to_string__vector( $r->requests, '\shopndrop_web_protocol\to_string__ShoppingRequestInfo' ); // Array

    return $res;
}

function to_string__GetShoppingListWithTotalsRequest( & $r )
{
    $res = "";

    // base class
    $res .= \shopndrop_protocol\to_string__Request( $r );

    $res .= " shopping_list_id=" . \basic_parser\to_string__int( $r->shopping_list_id );

    return $res;
}

function to_string__GetShoppingListWithTotalsResponse( & $r )
{
    $res = "";

    // base class
    $res .= \generic_protocol\to_string__BackwardMessage( $r );

    $res .= " shopping_list=" . to_string__ShoppingListWithTotals( $r->shopping_list );

    return $res;
}

function to_string__GetDashScreenUserRequest( & $r )
{
    $res = "";

    // base class
    $res .= \shopndrop_protocol\to_string__Request( $r );

    $res .= " user_id=" . \basic_parser\to_string__int( $r->user_id );
    $res .= " position=" . \shopndrop_protocol\to_string__GeoPosition( $r->position );

    return $res;
}

function to_string__GetDashScreenUserResponse( & $r )
{
    $res = "";

    // base class
    $res .= \generic_protocol\to_string__BackwardMessage( $r );

    $res .= " dash_screen=" . to_string__DashScreenUser( $r->dash_screen );

    return $res;
}

function to_string__GetDashScreenShopperRequest( & $r )
{
    $res = "";

    // base class
    $res .= \shopndrop_protocol\to_string__Request( $r );

    $res .= " user_id=" . \basic_parser\to_string__int( $r->user_id );
    $res .= " position=" . \shopndrop_protocol\to_string__GeoPosition( $r->position );

    return $res;
}

function to_string__GetDashScreenShopperResponse( & $r )
{
    $res = "";

    // base class
    $res .= \generic_protocol\to_string__BackwardMessage( $r );

    $res .= " dash_screen=" . to_string__DashScreenShopper( $r->dash_screen );

    return $res;
}

// generic

function to_string( $obj )
{
    $handler_map = array(
        // objects
        'shopndrop_web_protocol\ProductItemWithId'         => 'to_string__ProductItemWithId',
        'shopndrop_web_protocol\ShoppingItemWithProduct'         => 'to_string__ShoppingItemWithProduct',
        'shopndrop_web_protocol\ShoppingListWithProduct'         => 'to_string__ShoppingListWithProduct',
        'shopndrop_web_protocol\ShoppingListWithTotals'         => 'to_string__ShoppingListWithTotals',
        'shopndrop_web_protocol\RideSummaryWithShopper'         => 'to_string__RideSummaryWithShopper',
        'shopndrop_web_protocol\RideWithId'         => 'to_string__RideWithId',
        'shopndrop_web_protocol\ShoppingRequestInfo'         => 'to_string__ShoppingRequestInfo',
        'shopndrop_web_protocol\AcceptedOrderUser'         => 'to_string__AcceptedOrderUser',
        'shopndrop_web_protocol\AcceptedOrderShopper'         => 'to_string__AcceptedOrderShopper',
        'shopndrop_web_protocol\DashScreenUser'         => 'to_string__DashScreenUser',
        'shopndrop_web_protocol\DashScreenShopper'         => 'to_string__DashScreenShopper',
        // messages
        'shopndrop_web_protocol\GetProductItemListRequest'         => 'to_string__GetProductItemListRequest',
        'shopndrop_web_protocol\GetProductItemListResponse'         => 'to_string__GetProductItemListResponse',
        'shopndrop_web_protocol\GetShoppingRequestInfoRequest'         => 'to_string__GetShoppingRequestInfoRequest',
        'shopndrop_web_protocol\GetShoppingRequestInfoResponse'         => 'to_string__GetShoppingRequestInfoResponse',
        'shopndrop_web_protocol\GetShoppingListWithTotalsRequest'         => 'to_string__GetShoppingListWithTotalsRequest',
        'shopndrop_web_protocol\GetShoppingListWithTotalsResponse'         => 'to_string__GetShoppingListWithTotalsResponse',
        'shopndrop_web_protocol\GetDashScreenUserRequest'         => 'to_string__GetDashScreenUserRequest',
        'shopndrop_web_protocol\GetDashScreenUserResponse'         => 'to_string__GetDashScreenUserResponse',
        'shopndrop_web_protocol\GetDashScreenShopperRequest'         => 'to_string__GetDashScreenShopperRequest',
        'shopndrop_web_protocol\GetDashScreenShopperResponse'         => 'to_string__GetDashScreenShopperResponse',
    );

    $type = get_class( $obj );

    if( array_key_exists( $type, $handler_map ) )
    {
        $func = '\\shopndrop_web_protocol\\' . $handler_map[ $type ];
        return $func( $obj );
    }

    return \shopndrop_protocol\to_string( $obj );
}

# namespace_end shopndrop_web_protocol


?>
